File: array_problems/countTargetChar/countTargetChar.py
class ArrayProblems:
    def CountTargetChar(self, characters, char):
        result = []
        characters = characters.lower()
        if len(characters) == 0:
            return False
        for i in range(len(characters)):
            value = characters[i]
            if value.lower() == char.lower():
                result.append(ord(value))
        return len(result)


array_problems = ArrayProblems()
numberChar = array_problems.CountTargetChar("Hello world, this is very basic for developers", 'S')
print(numberChar)

File: array_problems/countTargetChar/countTargetChar.php
<?php
class ArrayProblems {
     public function CountTargetChar($str, $targetChar) {
         $result = [];
         $str = strtolower($str);
         if(!strlen($str)) return false;
         for($i = 0; $i < strlen($str); $i++) {
             $char = $str[$i];
             if($char == $targetChar || ord(strtolower($char)) == ord(strtolower($targetChar))) {
                 $result[] = ord($char);
             }
         }
         return count($result);
     }
}
$arrayProblems = new ArrayProblems();
$numberChar = $arrayProblems->CountTargetChar("Hello world, this is very basic for developers", 'S');
echo $numberChar;